2023/P_U2/main.cpp

```diff
@@ -0,0 +1,104 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+#include <iomanip>
+using namespace std;
+
+struct Mokinys
+{
+    string vardas;
+    vector<string> zymes;
+    int laimejimuKiekis = 0;
+    int lygiujuKiekis = 0;
+    int pralaimejimuKiekis = 0;
+};
+
+void skaitymas(int &mokiniuKiekis, Mokinys mokiniai[])
+{
+    ifstream data("U2.txt");
+    int zymiuKiekis;
+    char ch[14];
+    string zyme;
+
+    data >> mokiniuKiekis >> zymiuKiekis;
+
+    for (int i = 0; i < mokiniuKiekis; i++)
+    {
+        data.ignore();
+        data.read(ch, 13);
+
+        mokiniai[i].vardas = ch;
+        for (int j = 0; j < zymiuKiekis; j++)
+        {
+            data >> zyme;
+            mokiniai[i].zymes.push_back(zyme);
+        }
+    }
+    data.close();
+}
+
+void mokiniuLyginimas(Mokinys &mok1, Mokinys mok2)
+{
+    for (int i = 0; i < mok1.zymes.size(); i++)
+    {
+        if (mok1.zymes[i] == mok2.zymes[i])
+        {
+            mok1.lygiujuKiekis++;
+        }
+        else if (mok1.zymes[i] == "A" && mok2.zymes[i] == "Z")
+        {
+            mok1.laimejimuKiekis++;
+        }
+        else if (mok1.zymes[i] == "A" && mok2.zymes[i] == "P")
+        {
+            mok1.pralaimejimuKiekis++;
+        }
+        else if (mok1.zymes[i] == "Z" && mok2.zymes[i] == "A")
+        {
+            mok1.pralaimejimuKiekis++;
+        }
+        else if (mok1.zymes[i] == "Z" && mok2.zymes[i] == "P")
+        {
+            mok1.laimejimuKiekis++;
+        }
+        else if (mok1.zymes[i] == "P" && mok2.zymes[i] == "A")
+        {
+            mok1.laimejimuKiekis++;
+        }
+        else if (mok1.zymes[i] == "P" && mok2.zymes[i] == "Z")
+        {
+            mok1.pralaimejimuKiekis++;
+        }
```

Comment:
DISCUSSION. Clear and explicit declaration, I like it! But won't it be enough to check only the winning hands, and consider all others as losing?